feat(asset-def): add support for hex and string on asset def

mduthey opened the pull request:
Add support on asset definition for hex and string on names

```tx3
asset MyTokenHex = 0x12345678.0xABC;
asset MyTokenAscii = 0x12345678.MY_TOKEN;
asset MyTokenString = 0x12345678."MY TOKEN";
```

scarmuega commented on crates/tx3-lang/src/ast.rs:
```diff
@@ -699,11 +699,18 @@ impl VariantCase {
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub enum AssetName {
+    Ascii(String),
+    String(StringLiteral),
+    HexString(HexStringLiteral),
+}
+
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
 pub struct AssetDef {
     pub name: String,
     pub policy: Option<HexStringLiteral>,
-    pub asset_name: Option<String>,
+    pub asset_name: Option<AssetName>,
     pub span: Span,
 }
```
I'm afraid that the Asset definition is becoming too opinionated and very coupled to a specific use-case.
I prefer if we turn into something that looks like this...

```pest
asset_def = {
    "asset" ~ identifier ~ "=" ~ data_expr ~ "." ~ data_expr ~ ";"
}
```

The analysis phase should be the one throwing an diagnostic error if any of the data_expr can't be coerced into bytes.

mduthey commented:
I updated the code to use DateExpr instead of custom definition for AssetDef.
- I removed `ascii_text` from tx3.pest as we don't need anymore (was used only for asset).
- Added new display type for DateExpr to add more information to new error.
- Added new error `InvalidTypeError` allowing to trigger type errors, used for asset currently:

Example of new error:
For tx3 asset definition:
```tx3
asset Fuel = 0xfc8ad4f84181b85dc04f7b8c2984b129284c4e272ef45cd6440575fd4655454c.FUEL;
```
where token `FUEL` is not longer valid (now should be an string with `"`) error will be 
```
InvalidType(InvalidTypeError { name: "Asset.Name", type: "Identifier(FUEL)", expected: "String or Hex", src: None, span: Span { dummy: false, start: 539, end: 543 } })
```
